x-pack/filebeat/input/azureeventhub/v2_input.go

```diff
@@ -0,0 +1,532 @@
+	//
+	// The processor uses the default start position for
+	// all partitions if there is no checkpoint information
+	// available from the storage account container.
+	defaultStartPosition := azeventhubs.StartPosition{}
+
+	switch config.ProcessorStartPosition {
+	case startPositionEarliest:
+		defaultStartPosition.Earliest = to.Ptr(true)
+	case startPositionLatest:
+		defaultStartPosition.Latest = to.Ptr(true)
+	}
+
+	return &azeventhubs.ProcessorOptions{
+		//
+		// The `LoadBalancingStrategy` controls how the
+		// processor distributes the partitions across the
+		// consumers.
+		//
+		// LoadBalancingStrategy offers multiple options:
+		//
+		// - Balanced
+		// - Greedy
+		//
+		// As of now, we only support the "balanced" load
+		// balancing strategy for retro compatibility with
+		// the old SDK.
+		//
+		LoadBalancingStrategy: azeventhubs.ProcessorStrategyBalanced,
+		UpdateInterval:        config.ProcessorUpdateInterval,
+		StartPositions: azeventhubs.StartPositions{
+			Default: defaultStartPosition,
+		},
+	}
+}
+
+// ensureContainerExists ensures the blob container exists.
+func (in *eventHubInputV2) ensureContainerExists(ctx context.Context, blobContainerClient *container.Client) error {
+	exists, err := in.containerExists(ctx, blobContainerClient)
+	if err != nil {
+		return fmt.Errorf("failed to check if blob container exists: %w", err)
+	}
+	if exists {
+		return nil
+	}
+
+	// Since the container does not exist, we create it.
+	r, err := blobContainerClient.Create(ctx, nil)
+	if err != nil {
+		// If the container already exists, we ignore the error.
+		var responseError *azcore.ResponseError
+		if !errors.As(err, &responseError) || responseError.ErrorCode != string(bloberror.ContainerAlreadyExists) {
+			return fmt.Errorf("failed to create blob container: %w", err)
+		}
+
+		in.log.Debugw("blob container already exists, no need to create a new one", "container", in.config.SAContainer)
+	}
+
+	in.log.Infow("blob container created successfully", "response", r)
+
+	return nil
+}
+
+// containerExists checks if the blob container exists.
+func (in *eventHubInputV2) containerExists(ctx context.Context, blobContainerClient *container.Client) (bool, error) {
+	// Try to access the container to see if it exists.
+	_, err := blobContainerClient.GetProperties(ctx, &container.GetPropertiesOptions{})
+	if err == nil {
+		in.log.Debugw("blob container already exists, no need to create a new one", "container", in.config.SAContainer)
+		return true, nil
+	}
+
+	var responseError *azcore.ResponseError
+	if errors.As(err, &responseError) && responseError.ErrorCode == string(bloberror.ContainerNotFound) {
+		return false, nil
+	}
+
+	return false, fmt.Errorf("failed to check if blob container exists: %w", err)
+}
+
+// workersLoop starts a goroutine for each partition to process events.
+func (in *eventHubInputV2) workersLoop(ctx context.Context, processor *azeventhubs.Processor) {
+	for {
```

Comment:
I think you should introduce for select instead.

1 case should handle <-ctx.Done() and the default should handle the actual logic

Reply:
The Processor controls the exit from the loop:

```go
processorPartitionClient := processor.NextPartitionClient(ctx)
if processorPartitionClient == nil {
	// We break out from the for loop when `NextPartitionClient`
	// return `nil` (signals the processor has stopped).
	break
}
```

Do you think we should check the context directly instead of indirectly using the Processor?

Reply:
Yeah, makes sense. Resolving this.